hw/ip/rom_ctrl/data/rom_ctrl.hjson

```diff
@@ -0,0 +1,66 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  name: "rom_ctrl"
+  clock_primary: "clk_i"
+  reset_primary: "rst_ni",
+  regwidth: "32"
+  bus_interfaces: [
+    { protocol: "tlul", direction: "device", name: "regs" }
+    { protocol: "tlul", direction: "device", name: "rom" },
+  ],
+  alert_list: [
+    { name: "fatal"
+      desc: "A fatal error. Fatal alerts are non-recoverable and will be asserted until a hard reset."
+    }
+  ],
+  inter_signal_list: [
+    // Interface to memory configuration
+    { name:    "rom_cfg",
+      package: "prim_rom_pkg",
+      struct:  "rom_cfg",
+      act:     "rcv"
+      type:    "uni",
+    }
+  ],
+  registers: {
+    regs: [
+      { name: "FATAL_ALERT_CAUSE",
+        desc: '''
+          The cause of a fatal alert.
+
+          The bits of this register correspond to errors that can cause a fatal
+          alert. Software can read these bits to see what went wrong. Once set,
+          these bits cannot be cleared.
+        '''
+        swaccess: "ro",
+        hwaccess: "hwo",
+        fields: [
+          { bits: "0",
+            name: "integrity_error",
+            resval: 0,
+            desc: "Set on an integrity error from the register interface"
+          }
+
+          { bits: "1",
+            name: "dummy",
+            resval: 0,
+            desc: "Dummy index to prevent reggen from hiding the array."
+          }
+        ]
+      }
+    ],
+
+    rom: [
```

Comment:
one other thought, do we need to template this file so that a user can easily change the rom size from `top_earlgrey.hjson`?

This might answer my other question regarding why the second regfile is created, since it seems like the size is derived from there.

Reply:
Well, that's definitely a thing we *could* do. Since that introduces a whole load more pain, I'd rather not do it right now, especially since we don't currently have a need. I'm hopeful that the work @imphil's doing to improve the generated IP block infrastructure will make this cleaner in the future. And I'm also hopeful that I can avoid thinking about it until he's done :grin:

Reply:
The less templating the better -- I'm digging myself through what we currently have and (as you'll see in the PRs coming up reasonably soon), it's not a fun thing to look at. Please don't add to it unless we have a good reason.

Yes, these things will get easier, but the path to get there won't be pretty.